gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/GradebookPage.java

```diff
@@ -388,7 +389,15 @@ public String getCssClass() {
 
 				};
 
-				cols.add(column);
+				if (settings.isCategoriesEnabled()) {
+					// insert category column after assignments in that category
+					currentColumnIndex = currentColumnIndex + category.getAssignmentList().size();
+					cols.add(currentColumnIndex, column);
+					currentColumnIndex = currentColumnIndex + 1;
+				} else {
```

Comment:
Oo.. I replaced that really ugly IColumn sort and inline comparator with something a little simpler.. we can use the assignment counts to add the category average column at the right point in the columns array.  Hurray!